File: Python/413. Arithmetic Slices.py
class Solution:
    def numberOfArithmeticSlices(self, A: List[int]) -> int:
        res, count = 0, 1
        for i in range(1, len(A)-1):
            if A[i+1]-A[i] == A[i]-A[i-1]:
                res = res + count
                count += 1
            else:
                count = 1

        return res

File: 算法/C++/413. Arithmetic Slices.cpp
class Solution
{
public:
    int numberOfArithmeticSlices(vector<int> &A)
    {
        int res = 0, count = 1;
        if (A.size() < 3)
            return 0;
        for (int i = 1; i < A.size() - 1; i++)
        {
            if (A[i + 1] - A[i] == A[i] - A[i - 1])
            {
                res += count++; //arithmetic增加的个数符合等差数列
            }
            else
            {
                count = 1;
            }
        }
        return res;
    }
};